### PR TITLE
Fix: unsupported media type 

### PR DESCRIPTION
**Pull request by reliq**
Fix unsupported media type issue in certain situations such as posting direct messages.

ref: #295 

---

By introducing a _request_format_ option we can ensure requests are made in the correct formats, irrespective of desired response format.

### CODE REVIEW

File: src/Twitter.php [hunk 1]
```diff
@@ -313,10 +322,10 @@ private function formatResponse(Response $response, string $format)
             case self::RESPONSE_FORMAT_JSON:
                 return $body;
             case self::RESPONSE_FORMAT_ARRAY:
-                return json_decode($body, true);
+                return json_decode($body, true, 512, JSON_THROW_ON_ERROR);
```
Comment (atymic) [inline src/Twitter.php]: I guess for this we'll need to bump the min version to 7.3 (or maybe use a polyfill)?

File: src/Twitter.php [hunk 2]
```diff
@@ -325,16 +334,19 @@ private function formatResponse(Response $response, string $format)
...
      * @return mixed|string
      */
     private function request(string $url, array $parameters, string $method)
     {
-        $format = $parameters[self::KEY_FORMAT] ?? self::RESPONSE_FORMAT_OBJECT;
-        $requestOptions = $this->getRequestOptions($parameters, $method);
+        $responseFormat = $parameters[self::KEY_RESPONSE_FORMAT] ?? self::RESPONSE_FORMAT_OBJECT;
+        $requestFormat = $parameters[self::KEY_REQUEST_FORMAT] ?? $responseFormat;
+
+        unset($parameters[self::KEY_RESPONSE_FORMAT]);
```
Comment (atymic) [inline src/Twitter.php]: Do we need to unset the request format here?

File: src/Twitter.php (hunk 1 above)
Comment (reliq) [reply src/Twitter.php]: Yes. I'd prefer we bump the min version

File: src/Twitter.php (hunk 2 above)
Comment (reliq) [reply src/Twitter.php]: Not really but I did it so that we make no changes to the `$parameters` inside `getRequestOptions()`.

**Pull request merged**